message: support for marshalling embedded structs

seglberg opened the pull request:
Add support for marshalling (and unmarshalling) with embedded structs.
Embedded structs can use the new "inline" struct tag opt to signal that they
should be treated as inlined values.

```go
  type Embedded struct {
      Field string `vici:"field"`
  }

  type MyMessage struct {
      Embedded  `vici:",inline"`
      Other     string `vici:"other"`
  }
```

This allows using a common embedded struct for different message types which
share common fields.

Includes tests for both the tag parsing function (to ensure that skip still works as expected) and obviously for the inline marshal/unmarshal.

Let me know what you think!

enr0n commented:
This looks great, I think the code is good to go. I just have one request, and one clarifying question.

Request: can you please update the `MarshalMessage` and `UnmarshalMessage` comments so that the new `inline` option is documented?

Question: Marshaling embedded structs was previously supported intentionally, right? It would just be marshaled into a message with a sub-message (I realized, however, we don't appear to have any tests for this case). The commit message makes it seem like marshaling structs with embedded structs was not previously do-able in any fashion.

enr0n commented:
Also, was there any significant change in that force push I should be aware of? I did my review on the previous version.

seglberg commented:
No, there isn't any logic changes in the force push. Just pushed a change I forgot yesterday which combined setting the name in the messageTag:

```
mt := messageTag{name: opts[0]}
```

Good point on the embedded structs without being inlined. They should work if they're given a tag name. I can add a test for that as well just to be thorough, and update the commit message to be more succinct.

seglberg commented:
- Added a second commit with tests for marshalling embedded structs which have field name set on the vici struct tag (existing behavior).
- Touched up commit message to make it clear we're specifically adding inline support.
- Added brief documentation about inline opt to `MarshalMessage` and `UnmarshalMessage`.
